tests/test_main_window.py

```diff
@@ -67,12 +68,20 @@ def test_saving_mda(qtbot: "QtBot", main_window: MainWindow, T, C, splitC, Z):
 
         mda = None
 
-        @main_window._mmc.events.sequenceStarted.connect
+        mmc = main_window._mmc
+        # re-register twice to fully exercise the logic of the update
+        # functions - the initial connections are made in init
+        # then after that they are fully handled by the _update_mda_engine
+        # callbacks
```

Comment:
true, though perhaps we should have used `self.register_mda_engine(MDAEngine(self))` instead of `self._mda_engine = ... ` in the `__init__` method over in pymmcore_plus, to avoid any potential diversion between init and method-based registration in the future?